### ISSUE TITLE
Buying ammo from caravan - caravan pawns reserve ammo for pickup

### ISSUE THREAD

**Issue by rhonse** [NoImageAvailable/CombatExtended#301]
Observed 0.17.1.4c, with CE Guns 1.4. Also running hand me that brick, rimsenal - rimhair, and psychology. Bought 7.62x51 FMJ from a caravan. A caravan pawn immediately reserved the ammunition for pickup. I used one of my pawns to prioritize hauling, and after it was placed in my stockpile the caravan pawn no longer tried to pick it up.

**Comment by NoImageAvailable** [comment]
So a friendly NPC from the caravan tried to pick up the ammo? Do you have the auto-pick up ammo option enabled in the mod settings?

**Comment by rhonse** [comment]
Yes. A friendly caravan NPC tried to pick up the ammo I bought from his caravan that matched the weapon he was carrying.

The auto-pick up ammo option is enabled.

I haven't seen another caravan yet, but I'll try to reproduce it when I do.

**Comment by Alicecomma** [comment]
"Traders" are exempt from TakeAndEquip, but caravans also have CaravanGuard which isn't marked as trader and will therefore pick up ammo. It makes sense for them to pick up ammo in some cases (combat), so they will need to be an exception in the TakeAndEquip system.
